src/form/internal/FormButton.ts

```diff
@@ -0,0 +1,35 @@
+import {FormElementType, IFormElement, IFormElementDesc, IFormParent} from '../interfaces';
+import * as d3 from 'd3';
+import {EventHandler} from 'phovea_core/src/event';
+
+export interface IButtonElementDesc extends IFormElementDesc {
+  iconClasses: string;
+}
+
+export default class FormButton extends EventHandler implements IFormElementDesc, IFormElement {
+  private $button: d3.Selection<HTMLButtonElement>;
+  private $node;
+
+  readonly type: FormElementType.BUTTON;
+  readonly id: string;
+
+  constructor(readonly parent: IFormParent, readonly $parent, readonly desc: IButtonElementDesc) {
+    super();
+    this.id = desc.id;
+    this.$node = $parent.append('div').classed('form-group', true);
+    this.build();
+  }
+
+  /**
+   * Set the visibility of an form element - needed by IFormElement
+   * @param visible
+   */
+  setVisible(visible: boolean) {
+    this.$node.classed('hidden', !visible);
+  }
+
+  build() {
+    this.$button = this.$node.append('button').classed('btn btn-primary', true);
```

Comment:
yeah, after I wrote this I tried to do it like
```
this.$button.on('click.propagate', () => {
    this.fire('click', this.$button);
});
```

which I saw in FormInputText

But I like passing the change listener, like you suggested, more